src/Paket.Core/Versioning/SemVer.fs

```diff
@@ -47,42 +53,72 @@ type PreRelease =
     { Origin : string
       Name : string
       Values : PreReleaseSegment list }
-    
+      
     static member TryParse (str : string) = 
-        if String.IsNullOrEmpty str then None 
+        if String.IsNullOrEmpty str then None
         else
-            let name = Regex("^(?<name>[a-zA-Z]+)").Match(str).Value
-
+            let getName fromList =
+                match fromList with
+                | AlphaNumeric(a)::_ -> a
+                | _::AlphaNumeric(a)::_ -> a // fallback to 2nd
+                | _ -> ""
+                
             let parse segment =
                 match bigint.TryParse segment with
-                | true, bint -> Numeric bint
-                | false, _ -> AlphaNumeric segment
+                | true, number when number >= 0I -> Numeric number
+                | _ -> AlphaNumeric segment
+                
+            let notEmpty = StringSplitOptions.RemoveEmptyEntries
+            let name, values = 
+                match str.Split([|'.'|],notEmpty) with
```

Comment:
heuristics: versions with dots in prerelease are semver2, which allows hyphen as part of identifiers; 
only when having no dots we try to interpret the embedded numbers and separate "prerelease name" via regex